File: app/src/main/java/com/hfad/bitsandpizzas/Java/PastaFragment.java
package com.hfad.bitsandpizzas.Java;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.hfad.bitsandpizzas.R;

/**
 * A simple {@link Fragment} subclass.
 */
public class PastaFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        RecyclerView pizzaRecycler = (RecyclerView) inflater.inflate(
                R.layout.fragment_pizza, container, false);

        String[] pizzaNames = new String[Pasta.pastas.length];
        for (int i = 0; i < pizzaNames.length; i++) {
            pizzaNames[i] = Pasta.pastas[i].getName();
        }

        int[] pizzasImages = new int[Pasta.pastas.length];
        for (int i = 0; i < pizzasImages.length; i++) {
            pizzasImages[i] = Pasta.pastas[i].getImageResourceId();
        }

        CaptionedImagesAdapter adapter = new CaptionedImagesAdapter(pizzaNames, pizzasImages);
        pizzaRecycler.setAdapter(adapter);
        GridLayoutManager layoutManager = new GridLayoutManager(getActivity(), 2);
        pizzaRecycler.setLayoutManager(layoutManager);

        adapter.setListener(new CaptionedImagesAdapter.Listener() {
            @Override
            public void onClick(int position) {
                Intent intent = new Intent(getActivity(), PizzaDetailActivity.class);
                intent.putExtra(PizzaDetailActivity.EXTRA_PIZZA_ID, position);
                getActivity().startActivity(intent);
            }
        });
        return pizzaRecycler;
    }
}

File: app/src/main/java/com/hfad/bitsandpizzas/Kotlin/PastaFragment.kt
package com.hfad.bitsandpizzas.Kotlin


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.ListFragment
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.hfad.bitsandpizzas.Java.Pasta.pastas
import com.hfad.bitsandpizzas.R

/**
 * A simple [Fragment] subclass.
 */
class PastaFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val pizzaRecycler = inflater.inflate(R.layout.fragment_pizza, container, false) as RecyclerView

        val pizzaNames = arrayOfNulls<String>(pastas.size)
        for (it in pizzaNames.indices) {
            pizzaNames[it] = pastas[it].name
        }

        val pizzasImages = arrayOfNulls<Int>(pastas.size)
        for (it in pizzasImages.indices) {
            pizzasImages[it] = pastas[it].imageResourceId
        }

        val adapter = CaptionedImagesAdapter(pizzaNames, pizzasImages)
        pizzaRecycler.adapter = adapter
        val layoutManager = GridLayoutManager(activity, 2)
        pizzaRecycler.layoutManager = layoutManager
        return pizzaRecycler
    }
}
